design/open/lakefs-kv-execution-plan.md

```diff
@@ -0,0 +1,72 @@
+# lakeFS on KV - Execution Plan
+
+Key/value data storage package (kv) that will supply key/value access to data to replace the current storage and locking done by Postgres.
+High level overview of the interface described in https://github.com/treeverse/lakeFS/blob/ed55edc2c24b24ed4e6fcccad1ae95844d38b9ad/design/open/metadata_kv/index.md. The package will support multiple back-end implementations.
+
+## Transform from using db to kv package
+
+High level interface described in the kv proposal, we should consider taking the DynamoDB interface to address the way we communicate `set-if` functionality while marshal the same value without fetching data first.
+
+Key/value format can be done using ProtoBuf / JSON.
+
+- key format based on path - a format with escaping should be supported to enable easy way to encode values as part of the key
+- each model should include version as part of the model information
+- migrate from old model to new one without downtime
+
+## Per package changes
+
+For the following packages we should handle to move from 'db' package to 'kv'.
+During development it should enable a configuration flag to switch between using kv and db implementation.
+Any move transition should support migration from old Postgres data to new Postgres data format.
```

Comment:
What's a `move transition`? Are you referring to `new Postgres data format` per package?

Reply:
Regarding this feature flag - is it only to control the way we swotch from the old DB usage to the new one, or is it intended to allow us to go back too?
If the later, does that mean that for the entire process we should maintain the entire  data on both?

Reply:
- Remove the extra description about the transition - it is the migration from db postgres to kv based postgres.
- One we - just a way for us to develop the kv capability while db is still the version we used in the project until feature is complete. It should enable us to develop and test kv while we deliver a new release to the project, until feature complete.
-